### ISSUE TITLE
Unhandled exception on parse

### ISSUE THREAD

**Issue by daFreeMan** [rubberduck-vba/Rubberduck#2716]
RD 2.0.11 Excel2010 Win7

After initializing a new repo (via SC), I clicked the `Pending` button on the RD tool bar to kick off the initial parse. I got a message from .NET framework:
![rd unhandled exception](https://cloud.githubusercontent.com/assets/11889733/23167496/a8ed40c2-f812-11e6-9cf4-51b291fff0c8.PNG)
```
See the end of this message for details on invoking 
just-in-time (JIT) debugging instead of this dialog box.

************** Exception Text **************
System.NullReferenceException: Object reference not set to an instance of an object.
   at Rubberduck.UI.SourceControl.BranchesViewViewModel.RefreshView() in c:\Users\Mathieu\Documents\GitHub\Rubberduck\RetailCoder.VBE\UI\SourceControl\BranchesViewViewModel.cs:line 54
   at Rubberduck.UI.SourceControl.SourceControlViewViewModel.Refresh() in c:\Users\Mathieu\Documents\GitHub\Rubberduck\RetailCoder.VBE\UI\SourceControl\SourceControlViewViewModel.cs:line 897
   at Rubberduck.UI.Command.MenuItems.UiDispatcher.<>c__DisplayClass1.<InvokeAsync>b__0(Object x) in c:\Users\Mathieu\Documents\GitHub\Rubberduck\RetailCoder.VBE\UI\Command\MenuItems\UiDispatcher.cs:line 29
```

Full RD log: 
[RubberduckLog.txt](https://github.com/rubberduck-vba/Rubberduck/files/790261/RubberduckLog.txt)

I closed Excel, reopened, then kicked off the initial parse with no other intervening activity and the parse completed successfully with no errors.

**Comment by retailcoder** [comment]
This is a recurring error, I think there must be 10 open issues with it.

>   at Rubberduck.UI.SourceControl.BranchesViewViewModel.RefreshView() in c:\Users\Mathieu\Documents\GitHub\Rubberduck\RetailCoder.VBE\UI\SourceControl\BranchesViewViewModel.cs:line 54

Looks like it's this line:

````csharp
CurrentBranch = Provider.CurrentBranch.Name;
````

Either the `Provider` or its `CurrentBranch` is `null` when that code runs. This is a serious issue in the `BranchesViewViewModel`. The `CurrentBranch` property on `GitProvider` is really `CurrentBranchName`, as it's returning a `string` and it's hiding the base class' `CurrentBranch` property which returns an actual `IBranch` object... but then it's hard to justify why `GitProvider` would be hiding any members if the viewmodel is written against the `ISourceControlProvider` interface, assuming LSP isn't taking too much of a beating.

I'd add a null-check on that line, but then it's not clear whether that would break something else, given . The SC viewmodels need to be thoroughly looked at, there's technical debt to be paid there too.

**Comment by daFreeMan** [comment]
Sorry, failed to research before reporting. I could have added a "Me Too!" post elsewhere. Of course, not all of us would immediately recognize duplicate errors... :/

**Comment by mansellan** [comment]
Log shows:

`Rubberduck.UI.SourceControl.SourceControlViewViewModel;Displaying Error with title 'Unable to perform initial commit.' and message 'Configuration value 'user.name' is missing or invalid.';`

This should be fixed now by #3401.
